mlrun/api/api/endpoints/submit.py

```diff
@@ -17,14 +18,25 @@
 @router.post("/submit_job")
 @router.post("/submit_job/")
 async def submit_job(
-    request: Request, db_session: Session = Depends(deps.get_db_session)
+    request: Request,
+    username: Optional[str] = Header(None, alias='x-remote-user'),
+    db_session: Session = Depends(deps.get_db_session),
 ):
     data = None
     try:
         data = await request.json()
     except ValueError:
         log_and_raise(HTTPStatus.BAD_REQUEST, reason="bad JSON body")
 
+    # enrich job task with the username from the request header
+    if username:
+        # if task is missing, we don't want to create one
+        if 'task' in data:
+            labels = data['task'].setdefault('metadata', {}).setdefault('labels', {})
+            # TODO: remove this duplication
+            labels.setdefault('v3io_user', username)
```

Comment:
@Hedingber i guess that setting v3io_user would depend on having an v3io access key set (in the header)

Reply:
Talked offline, there is a bit of "mishmash" - this `v3io_user` is what used for substitution for things like `{{run.user}}` templating, but the access key is coming from the function itself (its mount) so no need to do this check here